hierarchy/provider.cpp

```diff
@@ -0,0 +1,27 @@
+#include "provider.h"
+#include <iostream>
+
+provider::provider():person()
+{
+}
+
+provider::provider(const provider & toAdd):person(toAdd)
+{
+	consultNum = toAdd.consultNum;
+	weeklyFee = toAdd.weeklyFee;
```

Comment:
Providers should not be storing this, this number should be dynamically generated whenever a report is created.